src/debugger.ts

```diff
@@ -0,0 +1,45 @@
+import {
+	DebugSession,
+	InitializedEvent, TerminatedEvent, StoppedEvent, BreakpointEvent, OutputEvent, Event,
+	Thread, StackFrame, Scope, Source, Handles, Breakpoint
+} from 'vscode-debugadapter';
+import {DebugProtocol} from 'vscode-debugprotocol';
+import {readFileSync} from 'fs';
+import {basename} from 'path';
+
+
+/**
+ * This interface should always match the schema found in the mock-debug extension manifest.
+ */
+export interface LaunchRequestArguments extends DebugProtocol.LaunchRequestArguments {
+	/** An absolute path to the program to debug. */
+	program: string;
+	/** Automatically stop target after launch. If not specified, target does not stop. */
+	stopOnEntry?: boolean;
+}
```

Comment:
Not actually needed however, I thought I would need to at first but when I realized how the protocol was setup it was simpler overall to to write everything in Rust.